processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -119,4 +128,170 @@ public Filter apply(@Nullable String input)
 
     return filter;
   }
-}
+
+  public static Filter[] partitionWithBitmapSupport(Filter current)
+  {
+    return partitionFilterWith(
+        current, new Predicate<Filter>()
+        {
+          @Override
+          public boolean apply(Filter input)
+          {
+            return input.supportsBitmap();
+          }
+        }
+    );
+  }
+
+  private static Filter[] partitionFilterWith(Filter current, Predicate<Filter> predicate)
+  {
+    if (current == null) {
+      return null;
+    }
+    current = pushDownNot(current);
+    current = convertToCNF(current);
```

Comment:
means Conjunctive Normal Form, which is expressions having AND clauses only in top.